lightning/src/routing/router.rs

```diff
@@ -357,14 +357,25 @@ impl Writeable for Route {
 	fn write<W: crate::util::ser::Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
 		write_ver_prefix!(writer, SERIALIZATION_VERSION, MIN_SERIALIZATION_VERSION);
 		(self.paths.len() as u64).write(writer)?;
+		let mut blinded_tails = Vec::new();
 		for path in self.paths.iter() {
 			(path.hops.len() as u8).write(writer)?;
-			for hop in path.hops.iter() {
+			for (idx, hop) in path.hops.iter().enumerate() {
 				hop.write(writer)?;
+				if let Some(blinded_tail) = &path.blinded_tail {
+					if blinded_tails.is_empty() {
+						blinded_tails = Vec::with_capacity(path.hops.len());
+						for _ in 0..idx {
+							blinded_tails.push(None);
+						}
+					}
+					blinded_tails.push(Some(blinded_tail));
+				} else if !blinded_tails.is_empty() { blinded_tails.push(None); }
```

Comment:
Originally I required all-or-nothing for `Path` `BlindedTail`s, but ended up allowing it because even though we'll never construct such `Route`'s, we'd have no issue paying them. See https://github.com/lightningdevkit/rust-lightning/pull/2146#discussion_r1160865591